app/controllers/organizations_controller.rb

```diff
@@ -31,6 +31,9 @@ def create
 
     respond_to do |format|
       if @organization.save
+        if params[:organization][:attachment]
+          add_attachment
```

Comment:
This code is duplicated in `update` and `create` actions. Also it's not approach: you save organization object two times (it's two requests to DB), also there is a problem with situation if attachment will be not saved and organization already saved... This should be handled together: if any error appear's with any attribute then saving should be interrupted, and saved only if all is valid.